deep-prove/src/middleware/v2.rs

```diff
@@ -0,0 +1,44 @@
+use base64::{Engine, prelude::BASE64_STANDARD};
+use ff_ext::GoldilocksExt2;
+use mpcs::{Basefold, BasefoldRSParams, Hasher};
+use serde::{Deserialize, Serialize};
+use zkml::{Proof as ZkmlProof, inputs::Input, quantization::ScalingStrategyKind};
+
+/// A type of the proof for the `v2` of the protocol
+pub type Proof = ZkmlProof<GoldilocksExt2, Basefold<GoldilocksExt2, BasefoldRSParams<Hasher>>>;
+
+#[derive(Serialize, Deserialize)]
+pub struct ClientToGw {
+    /// The user-facing name of the submitted task.
+    pub pretty_name: String,
+
+    /// The ID of the model to use.
+    pub model_id: i32,
+
+    /// An array of inputs to run proving for
+    pub input: Input,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct GwToWorker {
+    /// The job ID to use when communicating with the gateway.
+    pub job_id: i64,
+
+    /// The base64-encoded model - tests on random binary data show that base64
+    /// encoding is 30% the size of classic array-of-bytes serde_json encoding.
+    pub model: String,
+
+    /// An array of inputs to run proving for
+    pub input: Input,
+}
+impl From<GwToWorker> for super::v1::DeepProveRequest {
```

Comment:
`v1::DeepProveRequest` is a good base to interact with the prover until we develop a smarter shared model database.